uta_tools/data_sources/mane_transcript.py

```diff
@@ -353,16 +364,39 @@ def _validate_index(self, ac: str, pos: Tuple[int, int],
         """
         start_pos = pos[0] + coding_start_site
         end_pos = pos[1] + coding_start_site
-        if self.seqrepo_access.is_valid_input_sequence(
-                ac, start_pos, end_pos)[0]:
+        if self.seqrepo_access.get_reference_sequence(ac, start_pos, end_pos,
+                                                      residue_mode=ResidueMode.INTER_RESIDUE)[0]:  # noqa E501
             return True
         else:
             return False
 
+    def _get_prioritized_transcripts_from_gene(self,
+                                               df: pd.core.frame.DataFrame) -> List:
+        """Sort and filter transcripts from gene to get priority list
+
+        :param pd.core.frame.DataFrame df: Data frame containing transcripts from gene
+            data
+        :return: List of prioritized transcripts for a given gene. Sort by latest
+            assembly, longest length of transcript, with first-published transcripts
+            breaking ties. If there are multiple transcripts for a given accession, the
+            most recent version of a transcript associated with an assembly will be kept
+        """
+        copy_df = df.copy(deep=True)
+        copy_df = copy_df.drop(columns="alt_ac").drop_duplicates()
+        copy_df["ac_no_version_as_int"] = copy_df["tx_ac"].apply(lambda x: int(x.split(".")[0].split("NM_00")[1]))  # noqa: E501
+        copy_df["ac_version"] = copy_df["tx_ac"].apply(lambda x: x.split(".")[1])
+        copy_df = copy_df.sort_values(["ac_no_version_as_int", "ac_version"],
+                                      ascending=[False, False])
+        copy_df = copy_df.drop_duplicates(["ac_no_version_as_int"], keep="first")
+        copy_df.loc[:, "len_of_tx"] = copy_df.loc[:, "tx_ac"].apply(lambda ac: len(self.seqrepo_access.get_reference_sequence(ac)[0]))  # noqa: E501
+        copy_df = copy_df.sort_values(
+            ["len_of_tx", "ac_no_version_as_int"], ascending=[False, True])
+        return list(copy_df["tx_ac"])
+
     async def get_longest_compatible_transcript(
             self, gene: str, start_pos: int, end_pos: int,
             start_annotation_layer: str, ref: Optional[str] = None,
-            residue_mode: str = ResidueMode.RESIDUE
+            residue_mode: str = ResidueMode.RESIDUE, mane_transcripts: Set = set()
```

Comment:
I think the recommended pattern here is to make `mane_transcript` Optional/None by default, and if it's None, change it to a set() in the method body

https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
I think the recommendation is if you are mutating the default parameter. With this parameter, we're not mutating it at all. We're just reading it.

Reply:
It's an easy change though. So I'll go ahead and set it to None

Reply:
oh I see what you mean, yeah should be fine